File: server/DigitalTwins.DAL/Entities/Device.cs
using DigitalTwins.Common.Enums;

namespace DigitalTwins.DAL.Entities;

public class Device
{
    public long Id { get; set; }
    
    public Guid UGuid { get; set; }

    public long TemplateId { get; set; }

    public string Name { get; set; } = string.Empty;
    
    public string AzureDigitalTwinUrl { get; set; } = string.Empty;

    public long CreatedBy { get; set; }

    public Status Status { get; set; }

    public string TopicName { get; set; } = string.Empty;

    public Template Template { get; set; } = null!;

    public User User { get; set; } = null!;

    public IEnumerable<WidgetDevice> WidgetDevices { get; set; } = null!;
    
    public IEnumerable<Statistic> Statistics { get; set; } = null!;
    
    public IEnumerable<ErrorLog> ErrorLogs { get; set; } = null!;
}

File: client/src/types/device.ts
import { User } from '@@types/user';
import { Hardware } from '@@types/hardware';

export enum Status {
  off,
  on,
}

export interface Device {
  id: number;
  uGuid: string;
  topicName: string;
  azureDigitalTwinUrl: string;
  templateId: number;
  template: {
    id: number;
    hardware: Hardware;
    name: string;
  };
  user: User;
  name: string;
  createdBy: User['id'];
  status: Status;
}

export interface CreateDeviceArgs {
  name: Device['name'];
  templateId: number;
  userId: User['id'];
}

export interface DeviceStatLog {
  id: number;
  statsTime: string;
  uptime: number;
  heapUsage: number;
  rssi: number;
  deviceId: Device['id'];
}

export interface DeviceErrorLog {
  id: number;
  name: string;
  description: string;
  dateTime: string;
  deviceId: Device['id'];
}
